### ISSUE TITLE
Installation instructions

### ISSUE THREAD

**Issue by vindarel** [fonol/parrot#1]
Hello, installation instructions would be a plus :)

I can install Tauri I suppose (which I don't know yet), then is there specific commands?

Best,

**Comment by fonol** [comment]
Hi, 
good point! Maybe I hesitated because I fear that at this stage, the project will only disappoint anyone who gets it running, because so much essential stuff is still missing. 
But anyway:
1. You need a working Rust installation and [Tauri](https://tauri.app/v1/guides/getting-started/prerequisites)
2. Node.js to use the tauri-cli toolset

Once you get the project to build, you can install the tauri-cli tools by running `npm install` in the project's root.
At this point, you should be able to run the project by running `npm run dev` in the project's root. This uses Tauri's CLI tools to provide a hot reloading of Rust code, so if you have the project running and change some Rust source code, it should automatically rebuild. 
During development, Tauri also needs something to serve your frontend stuff (i.e. everything in `/parrot_ui`).
See here: [development cycle](https://tauri.app/v1/guides/development/development-cycle)
Since I already use VSCode, for that part I just use the [Live Server](https://marketplace.visualstudio.com/items?itemName=ritwickdey.LiveServer) extension, since it requires basically no setup.
Any other dev server that can serve the content of the `/parrot_ui` folder will do though.
If you use the Live Server extension, you have to tell it what to use as root folder: Inside `<project_root>/.vscode/settings.json` add the following: 
```
    "liveServer.settings.multiRootWorkspaceName": "parrot_ui",
    "liveServer.settings.root": "/parrot_ui",
```
That should be it.
1. `npm run dev` inside the project's root will start the application, using the tauri cli.
2. Start your dev server (e.g. for VSCode and Live Server, you can right click on `/parrot_ui/index.html` and choose `Open with Live Server`, which will also open the app in a browser tab that you cannot use unfortunately due to the missing Tauri stuff). 

If you use a dev server that monitors your files for changes, you should now have a fully reloading app from the Rust and from the frontend side! :)

To build for release, run `npm run build:release` (if unsure about the commands, just look at the `package.json`).

Hope that helps! Let me know if it works for you!

**Comment by vindarel** [comment]
Thanks. I installed the Rust toolchain with the curl command, but I could probably have gone with `apt install rustc cargo`.

In `/parrot_ui` I started a local server with `python -m SimpleHTTPServer 5500` and I started Tauri with `npm run dev`. 

I got the Parrot UI, asking the paths to sbcl and the sbcl core (I gave it `/usr/local/lib/sbcl/sbcl.core`), but it couldn't connect to Sly, even though it looks like it was started, when looking at the Lisp stacktrace from the editor:

    ;; Slynk started at port: 4005.

maybe, only maybe, the front was too slow and the server didn't wait long enough? ("Could not connect to Slynk server over socket 127.0.0.1:4005. Retry 5 of 5 in 500ms...") (and the "restart repl" command crashes)

side note: I forgot what it is to wait for a build process^^

**Comment by fonol** [comment]
Hi,
thanks for the feedback!
Right of the top of my head, two things you could have a look at:

1. After you start the GUI, on the app's menu bar you can find the point `SBCL` > `View SBCL process output`. This just shows the standard output & error of the SBCL process that is started by the app and responsible for running the Slynk server. My guess would be you will see some kind of error in that output.

2. You can increase the number of connection retries in the file `parrot_rs/src/repl.rs`, line 640. But I don't think this is the culprit.

I think it would make sense if the connection to Slynk's server failed, to always directly show the SBCL process' output, to make troubleshooting easier. I also had this happen a lot during the start of development. 99/100 times it was Slynk/SBCL complaining about something.

**Comment by vindarel** [comment]
> 1. View SBCL process output

exactly, I see a list of warnings but in the end this is where I saw 

> ;; Slynk started at port: 4005.

!

But, on the terminal runnning "npm run dev", I see "Could not connect to Slynk server".

**Comment by fonol** [comment]
Hm okay, so you got neither in your terminal nor in the SBCL output dialog any errors except the "Could not connect to Slynk server"?
The app is starting the SBCL process like this `<abs path to SBCL executable (from settings)> --core <Abs path to SBCL core (from settings)> --load <Abs path to start-slynk.lisp>`. The first two come from your settings, the last one is calculated at runtime (start-slynk is in the project under `src-tauri/slynk/start-slynk.lisp`. If you just manually run this in a terminal with the correct paths filled in, does the Slynk server start normally?

**Comment by vindarel** [comment]
Exactly. 

The SBCL output dialog and this command give me the same output, which says Slynk was started:

    /usr/local/bin/sbcl --core /usr/local/lib/sbcl/sbcl.core  --load src-tauri/slynk/start-slynk.lisp

```
This is SBCL 2.1.5, an implementation of ANSI Common Lisp.
More information about SBCL is available at <http://www.sbcl.org/>.

SBCL is free software, provided as is, with absolutely no warranty.
It is mostly in the public domain; some portions are provided under
BSD-style licenses.  See the CREDITS and COPYING files in the
distribution for more information.
--- reading .sbclrc
--- asdf version: 3.3.4.15
WARNING:
   redefining SB-GRAY:STREAM-LINE-LENGTH (#<STANDARD-CLASS SB-GRAY:FUNDAMENTAL-CHARACTER-INPUT-STREAM>) in DEFMETHOD
WARNING:
   redefining EMACS-INSPECT (#<SB-PCL:SYSTEM-CLASS COMMON-LISP:T>) in DEFMETHOD
;; loading #P"/home/vince/.sly/fasl/1.0/sbcl-2.1.5-linux-x86-64/media/vince/93de6b92-7390-4253-bab1-425052ba7a10/home/vince/bacasable/parrot/src-tauri/target/debug/slynk/slynk-indentation.fasl"
WARNING: redefining SLYNK::SYMBOL-INDENTATION in DEFUN
WARNING: redefining SLYNK::MACRO-INDENTATION in DEFUN
;; loading #P"/home/vince/.sly/fasl/1.0/sbcl-2.1.5-linux-x86-64/media/vince/93de6b92-7390-4253-bab1-425052ba7a10/home/vince/bacasable/parrot/src-tauri/target/debug/slynk/slynk-stickers.fasl"
;; loading #P"/home/vince/.sly/fasl/1.0/sbcl-2.1.5-linux-x86-64/media/vince/93de6b92-7390-4253-bab1-425052ba7a10/home/vince/bacasable/parrot/src-tauri/target/debug/slynk/slynk-trace-dialog.fasl"
;; loading #P"/home/vince/.sly/fasl/1.0/sbcl-2.1.5-linux-x86-64/media/vince/93de6b92-7390-4253-bab1-425052ba7a10/home/vince/bacasable/parrot/src-tauri/target/debug/slynk/slynk-package-fu.fasl"
;; loading #P"/home/vince/.sly/fasl/1.0/sbcl-2.1.5-linux-x86-64/media/vince/93de6b92-7390-4253-bab1-425052ba7a10/home/vince/bacasable/parrot/src-tauri/target/debug/slynk/slynk-mrepl.fasl"
;; loading #P"/home/vince/.sly/fasl/1.0/sbcl-2.1.5-linux-x86-64/media/vince/93de6b92-7390-4253-bab1-425052ba7a10/home/vince/bacasable/parrot/src-tauri/target/debug/slynk/slynk-arglists.fasl"
;; Slynk started at port: 4005.
```
I pulled in your [last commits](https://github.com/fonol/parrot/compare/71591955c3...a6b3629d53).

I see:

> Failed to connect to Slynk server. 
> Connection refused (os error 111)

Probably I should check that I can connect to the Slynk server, started with the line above, from my editor. I currently have Slime. I'll try to check this.
